README.md

```diff
@@ -1,28 +1,123 @@
+
+## Add certificate to denylist
+The unsigned denylist is maintained at `config/certs.json`.
+
+Make a new entry using
+```shell
+# generate a denylist entry
+go run cli/dumpcert/main.go --cert <path-to-cert>
+```
+
+This will produce something like:
+```json
+{
+    "issuer": "CN=Nuts Development Network Root CA",
+    "serialnumber": "86909351591313461694061157857735660045310273765",
+    "jwkthumbprint": "xaVihytdhiID5wqCOsQbbb2_GuIb0r7m1GUgPIdBDjo",
+    "reason": "TODO: Give me one reason, give me just one reason why"
+}
+```
+
+Copy this into `config/certs.json` and set the reason for blocking the certificate.
+
+
+## Sign denylist
+The key to sign the denylist is stored in the `NutsDenylist` vault on 1password in the `PrivateKey` item.
+
+### 1password CLI (preferred)
+Install v2 of the 1password CLI by following the [instructions](https://developer.1password.com/docs/cli/get-started/) if needed.
+```shell
+# sign in to the correct account
+op signin
+
+# point the env variable to the correct value in 1password
+export DENYLIST_PRIVATEKEY_PEM="op://NutsDenylist/PrivateKey/password"
+
+# sign by letting 1password substitute the key
+op run make denylist
+```
+
+### Copy key to CLI (only if absolutely needed)
+If the 1password CLI is not an option for some reason, the denylist can also be signed after manually setting the env variable.
+```
+# get the key from 1password and subsitute below to sign the new denylist.
+DENYLIST_PRIVATEKEY_PEM="<private-key>" make denylist
+```
+
+
+## Publish denylist
+The previous step updated the signed denylist at `denylist/denylist.jws`. (DO NOT CHANGE THIS FILE LOCATION/NAME)
+Assuming the certificate to block was added to `certs/`, validate that the new denylist actually identifies the certificate.
+```shell
+# validate that the added certifcate is on the denylist
+make test
+```
+This loops over all the certificates in `certs/` and checks that they are on the denylist.
+
+If the test passes, merge the resulting changes to the `main` branch with a pull request.
+
+
+## Clear CDN cache
+The denylist is hosted on Github but distributed using [www.jsdelivr.com](https://www.jsdelivr.com), which is "A free CDN for open source projects".
+This CDN operates on top of multiple major CDNs and will provide us with a better uptime than Github can.
+The [default cache for jsdelivr](https://www.jsdelivr.com/documentation#id-caching) is 12 hours for files on Github branches.
+TODO: contact jsdelivr to see if/how we can speed this up.
+
+The denylist file can be found at https://cdn.jsdelivr.net/gh/nuts-foundation/denylist@main/denylist/denylist.jws. 
+This points to the `main` branch on the Github repo.
+
+
+## Troubleshooting
+- `error parsing key: failed to parse PEM encoded key: failed to decode PEM data`:  
+  PEM data in an env variable can be a bit iffy. The header and footer should be separated from the key using new line character `\n`.
```

Comment:
This is not an actual newline character, but rather a '\' character followed by an 'n' character. This is replaced with newline characters after the environment variable is read.